src/rest/mod.rs

```diff
@@ -3,7 +3,7 @@
 mod error;
 mod model;
 #[cfg(test)]
-mod tests;
+pub mod tests;
```

Comment:
I'd suggest using `pub(crate)` to make sure the tests module is only accessible within the crate, or even more restrictive `pub(in crate::ws::tests)`.

see https://doc.rust-lang.org/reference/visibility-and-privacy.html

Reply:
Ahh that's perfect, I didn't know it was possible. I changed it to `pub(crate)`; `pub(in crate::ws::tests)` required `crate::ws::tests` to be public as well.